src/party.cpp

```diff
@@ -432,8 +434,9 @@ bool Party::canUseSharedExperience(const Player* player) const
 		}
 	}
 
-	uint32_t minLevel = static_cast<int32_t>(std::ceil((static_cast<float>(highestLevel) * 2) / 3));
```

Comment:
There are lots of errors like this throughout the code. I'd even use `auto` here as the type is very explicit.